src/main/java/com/google/api/codegen/config/GapicProductConfig.java

```diff
@@ -945,38 +1126,38 @@ private static void createSingleResourceNameConfig(
                     + singleResourceNameConfig.getEntityId()));
       }
     } else {
-      singleResourceNameConfigsMap.put(
-          singleResourceNameConfig.getEntityId(), singleResourceNameConfig);
+      String configKey = singleResourceNameConfig.getEntityId();
+      configKey = StringUtils.prependIfMissing(configKey, prefixForMap);
+      singleResourceNameConfigsMap.put(configKey, singleResourceNameConfig);
     }
   }
 
-  // Construct a new SingleResourceNameConfig from the given Resource, and add the newly
+  // Construct a new FixedResourceNameConfig from the given Resource, and add the newly
   // created config as a value to the map param, keyed on the package-qualified entity_id.
-  private static void createSingleResourceNameConfig(
+  private static void createFixedResourceNameConfigFromProtoFile(
```

Comment:
I think this function is slightly misnamed, and could be simplified. The actual creation is just a one-liner. Most of the logic here is logic about _insertion_ into the map, and doesn't use the Resource or ProtoFile. So WDYT about pulling out the creation line, and making this an `insertFixedResourceNameConfig` method?

Reply:
I think then the insertion logic can actually be shared between "fromprotofile" and "fromgapic" cases (although not 100% sure).

Reply:
oh yes! done